src/ir/cfg_builder.cpp

```
@@ -345,6 +345,21 @@ static void add_cfg_nodes(CfgBuilder& builder, const Label& caller_label, const
     }
 }
 
+/// Resolve a LoadPseudo with map-by-index addressing to the concrete LoadMapFd or LoadMapAddress instruction.
+static Instruction resolve_map_by_index(const LoadPseudo& pseudo) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)